metricbeat/module/vsphere/host/host.go

```diff
@@ -92,83 +111,140 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+		"disk.usage.average":          &metricsVar.DiskUsage,
+		"disk.read.average":           &metricsVar.DiskRead,
+		"disk.write.average":          &metricsVar.DiskWrite,
+		"net.transmitted.average":     &metricsVar.NetTransmitted,
+		"net.received.average":        &metricsVar.NetReceived,
+		"net.packetsTx.summation":     &metricsVar.NetPacketTransmitted,
+		"net.packetsRx.summation":     &metricsVar.NetPacketReceived,
+		"net.errorsTx.summation":      &metricsVar.NetErrorsTransmitted,
+		"net.errorsRx.summation":      &metricsVar.NetErrorsReceived,
+		"net.multicastTx.summation":   &metricsVar.NetMulticastTransmitted,
+		"net.multicastRx.summation":   &metricsVar.NetMulticastReceived,
+		"net.droppedTx.summation":     &metricsVar.NetDroppedTransmitted,
+		"net.droppedRx.summation":     &metricsVar.NetDroppedReceived,
+	}
 
-		if hs.Summary.Host != nil {
-			networkNames, err := getNetworkNames(ctx, c, hs.Summary.Host.Reference())
-			if err != nil {
-				m.Logger().Debugf("error trying to get network names: %s", err.Error())
-			} else {
-				if len(networkNames) > 0 {
-					event["network_names"] = networkNames
-				}
-			}
+	var spec types.PerfQuerySpec
+	metricIDs := make([]types.PerfMetricId, 0, len(metricNames))
+
+	for _, metricName := range metricNames {
+		metric, exists := metrics[metricName]
+		if !exists {
+			m.Logger().Debug("Metric ", metricName, " not found")
+			continue
 		}
-		reporter.Event(mb.Event{
-			MetricSetFields: event,
+
+		metricIDs = append(metricIDs, types.PerfMetricId{
+			CounterId: metric.Key,
 		})
 	}
 
-	return nil
-}
-
-func getNetworkNames(ctx context.Context, c *vim25.Client, ref types.ManagedObjectReference) ([]string, error) {
-	var outputNetworkNames []string
-
 	pc := property.DefaultCollector(c)
+	for _, hs := range hst {
+		var networkRefs []types.ManagedObjectReference
+		for _, obj := range hs.Network {
+			if obj.Type == "Network" {
+				networkRefs = append(networkRefs, obj)
+			}
+		}
 
-	var hs mo.HostSystem
-	err := pc.RetrieveOne(ctx, ref, []string{"network"}, &hs)
-	if err != nil {
-		return nil, fmt.Errorf("error retrieving host information: %v", err)
-	}
+		var nets []mo.Network
+		if len(networkRefs) > 0 {
+			err = pc.Retrieve(ctx, networkRefs, []string{"name"}, &nets)
+			if err != nil {
+				m.Logger().Errorf("error retrieving network from host: %v", err)
+			}
+		}
 
-	if len(hs.Network) == 0 {
-		return nil, errors.New("no networks found")
-	}
+		var outputNetworkNames []string
+		for _, net := range nets {
+			name := strings.Replace(net.Name, ".", "_", -1)
+			outputNetworkNames = append(outputNetworkNames, name)
+		}
 
-	var networkRefs []types.ManagedObjectReference
-	for _, obj := range hs.Network {
-		if obj.Type == "Network" {
-			networkRefs = append(networkRefs, obj)
+		spec = types.PerfQuerySpec{
+			Entity:     hs.Reference(),
+			MetricId:   metricIDs,
+			MaxSample:  1,
+			IntervalId: 20, // right now we are only grabbing real time metrics from the performance manager
```

Comment:
@kush-elastic Can you pls tell why the interval has been chosen as 20 here 
Also, the relevance of MaxSample ?

Reply:
The `MaxSamples` setting controls how many metric samples you get back per interval. If you set it to `1`, you'll only get the latest metric. In our setup, with `IntervalId` set to `20`, we're grabbing real-time metrics because the server collects data every 20 seconds. So basically, we're just pulling the freshest 20-second metrics straight from the server.

Reply:
Check this out for more details: [vSphere Performance Data Collection](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-web-services-programming-guide/GUID-481756FA-9F0B-4768-8E3B-EAFCEEC908B2.html)